api/types.proto

```diff
@@ -713,3 +727,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
+
+	// Mode specifies how this secret should be exposed inside the task.
+	enum Mode {
+		FILE = 0;
```

Comment:
Let's add a `SYSTEM` mode, which means the secret will be available via direct access, usually at the agent-level, and won't be projected into file or env.